### PR TITLE
SAML build link

### PR DESCRIPTION
**Pull request by scruffystuffs**
## Description
Build SAML links if org uses SAML, if they don't or saml fails, use normal link

## Acceptance Criteria
* New link redirects thru SAML correctly (manually tested)
* New and old tests pass

## Testing plan
* Verify existing build links (if you don't already have SAML set up it can be difficult)

## Risks
Edge cases in v1 make this more risky than v2, but this is a best guess at handling them correctly.

### CODE REVIEW

File: api/fossa/locator.go [hunk 1]
```diff
@@ -82,9 +84,26 @@ func (l Locator) URL() string {
...
+	}
+
 	return server.ResolveReference(url).String()
 }
 
+// SamlURL constructs a SAML redirect URL for a given FOSSA URL
+func SamlURL(uri url.URL, org Organization) *url.URL {
+	samlBase, err := url.Parse("/account/saml/" + strconv.Itoa(org.OrganizationID))
+	if err != nil {
+		log.Fatalf("Invalid SAML URL construction: %s", err.Error())
```
Comment (zlav) [inline api/fossa/locator.go:99]: I think this should be `log.Warnf` to follow line 70. The project is still uploaded at this point if we get an error correct? If so I think we should make this a warning because `analyze` technically succeeded

Comment (scruffystuffs) [reply api/fossa/locator.go:99]: Technically, it doesn't matter, since this can't fail.  It's parsing a constant value and any integer.  But I can change it.

Comment (zlav) [reply api/fossa/locator.go:99]: Thats a good point... this is going to be nitpicky but can we change it `log.Warn` to follow our convention of not logging fatal unless absolutely necessary?

**Review by zlav — APPROVED** [review]
Looks great! +1 for the tests
